flink/flink.sh

```diff
@@ -70,39 +74,41 @@ function configure_flink() {
   # yarn.nodemanager.resource.cpu-vcores is correctly populated.
   local spark_executor_cores=$(\
     grep 'spark\.executor\.cores' /etc/spark/conf/spark-defaults.conf \
-    | tail -n1 | cut -d'=' -f2)
+      | tail -n1 | cut -d'=' -f2)
   local flink_taskmanager_slots="$(($spark_executor_cores * 2))"
   # local flink_taskmanager_slots="$(hdfs getconf \
   #   -confKey yarn.nodemanager.resource.cpu-vcores)"
 
-  # Determine the default parallelism
+  # Determine the default parallelism.
   local flink_parallelism=$(python -c \
-      "print ${num_taskmanagers} * ${flink_taskmanager_slots}")
+    "print ${num_taskmanagers} * ${flink_taskmanager_slots}")
 
   # Get worker memory from yarn config.
-  local worker_total_mem="$(hdfs getconf \
-    -confKey yarn.nodemanager.resource.memory-mb)"
+  local worker_total_mem="$(bdconfig get_property_value --configuration_file \
+    /etc/hadoop/conf/yarn-site.xml \
```

Comment:
This is strictly more brittle than using hdfs getconf as it depends on a hard-coded hadoop conf path. Why redirect stderr to /dev/null? If we want better error messaging, this should be captured with normal stderr.

Reply:
@dennishuo What do you think about using `bdutil` with a hard-coded configuration file rather than `hdfs getconf`?

Reply:
I agree hdfs getconf is probably better here; I think the confusion stems from my guidance in https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/master/CONTRIBUTING.md saying to use bdconfig instead of grep or sed against the XML files; in general bdconfig is still preferred vs string-replace hacks, but for cases where we can use Hadoop's tooling, that's indeed better than bdconfig. There are some edge cases where we might need to get values from specific files instead of relying on implicit import ordering of different xml files though, such as when we want to inspect distcp-default.xml and its overrides vs mapred-site.xml.

Reply:
Ah, that makes sense. I've updated the contributing guide to mention Hadoop tools in #189.